src/main/java/com/jayway/maven/plugins/android/phase09package/AarMojo.java

```diff
@@ -225,8 +225,16 @@ protected File createAarLibraryFile( File classesJar ) throws MojoExecutionExcep
 
     private void addR( JarArchiver jarArchiver ) throws MojoExecutionException
     {
-        File rFile = new File( project.getBuild().getDirectory() + "/R.txt" );
-        jarArchiver.addFile( rFile, "R.txt" );
+        final File rFile = new File( project.getBuild().getDirectory() + "/R.txt" );
+        if ( rFile.exists() )
+        {
+            jarArchiver.addFile( rFile, "R.txt" );
+            getLog().debug( "Packaging R.txt in AAR" );
+        }
+        else
```

Comment:
We should instead create an empty R.txt and add that in order to conform to the AAR file structure https://developer.android.com/studio/projects/android-library.html#aar-contents